apps/distgo/cmd/dist/dist_test.go

```diff
@@ -1179,3 +1186,169 @@ daemon: true
 		}
 	}
 }
+
+func TestDockerDist(t *testing.T) {
+	tmp, cleanup, err := dirs.TempDir("", "")
+	defer cleanup()
+	require.NoError(t, err)
+
+	for i, currCase := range []struct {
+		name            string
+		skip            func() bool
+		spec            func(projectDir string) []params.ProductBuildSpecWithDeps
+		preDistAction   func(projectDir string, buildSpec []params.ProductBuildSpecWithDeps)
+		skipBuild       bool
+		wantErrorRegexp string
+		validate        func(caseNum int, name string, projectDir string, cli *dockercli.Client)
+	}{
+		{
+			name: "docker dist",
+			spec: func(projectDir string) []params.ProductBuildSpecWithDeps {
+				fooSpecWithDeps, err := params.NewProductBuildSpecWithDeps(params.NewProductBuildSpec(
+					projectDir,
+					"foo",
+					git.ProjectInfo{
+						Version: "0.1.0",
+					},
+					params.Product{
+						Build: params.Build{
+							MainPkg: "./foo",
+							OSArchs: []osarch.OSArch{
+								{
+									OS:   "linux",
+									Arch: "amd64",
+								},
+							},
+						},
+						Dist: []params.Dist{
+							{
+								Info: &params.DockerDistInfo{
+									Repository: "foo",
+									Tag:        "0.1.0",
+									ContextDir: "foo/docker",
+									DistDeps: params.DockerDeps{
+										"bar": {
+											params.DockerDistType,
+										},
+									},
+								},
+							}},
+					},
+					params.Project{
+						GroupID: "com.test.group",
+					},
+				), nil)
+				require.NoError(t, err)
+				barSpecWithDeps, err := params.NewProductBuildSpecWithDeps(params.NewProductBuildSpec(
+					projectDir,
+					"bar",
+					git.ProjectInfo{
+						Version: "0.1.0",
+					},
+					params.Product{
+						Build: params.Build{
+							MainPkg: "./bar",
+							OSArchs: []osarch.OSArch{
+								{
+									OS:   "linux",
+									Arch: "amd64",
+								},
+							},
+						},
+						Dist: []params.Dist{
+							{
+								Info: &params.DockerDistInfo{
+									Repository: "bar",
+									Tag:        "0.1.0",
+									ContextDir: "bar/docker",
+								},
+							},
+						},
+					}, params.Project{
+						GroupID: "com.test.group",
+					},
+				), nil)
+				require.NoError(t, err)
+				return []params.ProductBuildSpecWithDeps{fooSpecWithDeps, barSpecWithDeps}
+			},
+			preDistAction: func(projectDir string, buildSpec []params.ProductBuildSpecWithDeps) {
+				gittest.CreateGitTag(t, projectDir, "0.1.0")
+			},
+			validate: func(caseNum int, name string, projectDir string, cli *dockercli.Client) {
+				filter := filters.NewArgs()
+				filter.Add("reference", "foo:0.1.0")
+				images, err := cli.ImageList(context.Background(), types.ImageListOptions{Filters: filter})
+				require.NoError(t, err)
+				require.True(t, len(images) > 0)
+				filter = filters.NewArgs()
+				filter.Add("reference", "bar:0.1.0")
+				images, err = cli.ImageList(context.Background(), types.ImageListOptions{Filters: filter})
+				require.NoError(t, err)
+				require.True(t, len(images) > 0)
```

Comment:
If we make the above change this will be slightly less of an issue, but I think it would still be nice to try to remove/clean the images created by the test at the end of the test (basically defer a call to remove the tags after they are created). Generally, tests should strive to leave the host system in the same state as before they were run.